lib/peep/buckets/custom.ex

```diff
@@ -7,62 +7,76 @@ defmodule Peep.Buckets.Custom do
   """
 
   defmacro __using__(opts) do
-    buckets =
-      Keyword.fetch!(opts, :buckets)
-      |> :lists.usort()
+    quote bind_quoted: [buckets: Keyword.fetch!(opts, :buckets)] do
+      @behaviour Peep.Buckets
 
-    unless Enum.all?(buckets, &is_number/1) do
-      raise ArgumentError, "expected buckets to be a list of numbers, got: #{buckets}"
-    end
+      require Peep.Buckets.Custom
 
-    number_of_buckets = length(buckets)
+      buckets = :lists.usort(buckets)
+      len = length(buckets)
 
-    quote do
-      @behaviour Peep.Buckets
+      unless Enum.all?(buckets, &is_number/1) do
+        raise ArgumentError, "expected buckets to be a list of numbers, got: #{buckets}"
+      end
 
       @impl true
       def config(_), do: %{}
 
       @impl true
-      def number_of_buckets(_), do: unquote(number_of_buckets)
+      def number_of_buckets(_), do: unquote(len)
 
       @impl true
-      unquote(bucket_for_ast(buckets))
-
-      @impl true
-      unquote(upper_bound_ast(buckets))
-    end
-  end
-
-  ## Bucket binary search
 
-  defp bucket_for_ast(buckets) do
-    int_buckets = int_buckets(buckets, nil, 0)
+      int_buckets =
+        buckets
+        |> Enum.map(&ceil/1)
+        |> Enum.with_index()
+        |> Enum.dedup_by(&elem(&1, 0))
+
+      float_buckets =
+        buckets
+        |> Enum.map(&(&1 * 1.0))
+        |> Enum.with_index()
+
+      if len <= 32 do
```

Comment:
I wonder if it should be configurable option to force one behaviour or another.

Reply:
Would you mind sharing your benchmark code and results?

Back when I was reviewing the PR that introduced binary search, I did not find a significant difference between linear search and binary search for small numbers of buckets.